app/common/directives/role-selector.directive.js

```diff
@@ -0,0 +1,32 @@
+module.exports = RoleSelectorDirective;
+
+RoleSelectorDirective.$inject = [];
+
+function RoleSelectorDirective() {
+    return {
+        restrict: 'E',
+        scope: {
+            model: '=',
+            title: '='
+        },
+        controller: RoleSelectorController,
+        template: require('./role-selector.html')
+    };
+}
+RoleSelectorController.$inject = ['$scope', 'RoleEndpoint'];
+
+function RoleSelectorController($scope, RoleEndpoint) {
+    $scope.addAllRoles = addAllRoles;
+
+    // getting available roles from api
+    RoleEndpoint.query().$promise.then(function (roles) {
```

Comment:
can we place this part inside an activate function:

activate();

function activate() {
   RoleEndpoint.query(...
}